src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -269,6 +435,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     void TermControl::_CloseSearchBoxControl(const winrt::Windows::Foundation::IInspectable& /*sender*/, RoutedEventArgs const& /*args*/)
     {
         _searchBox->Visibility(Visibility::Collapsed);
+        _searchState.reset();

```

Comment:
This is absolutely a future brainstorm, so feel free to ignore me.


But it might be neat to be able to set up a search string, then close the dialog, then like, hit <kbd>F3</kbd> to just perform a new search and hop to the search result. So you could search once, close the dialog, then do something in the terminal, and quickly repeat the search, if that makes sense.

That would involve a lot less "check if the search box is open" code. And here, we'd just keep the search state around. Though, that would mean we'd keep searching _always_ on output, which is maybe not what's expected. We'd need to not update the search results until the action was triggered.

\</brainstorm>

Reply:
Yeah.. I see your point. This can be a cool optimization to preserve the search results between dialog usages, but I guess we will still need to know if there is an active search or not to avoid being triggered on output.

Probably should still introduce as a follow-up